pkg/tsdb/mssql/macros.go

```diff
@@ -82,11 +82,12 @@ func (m *MsSqlMacroEngine) evaluateMacro(name string, args []string) (string, er
 		if len(args) == 0 {
 			return "", fmt.Errorf("missing time column argument for macro %v", name)
 		}
-		return fmt.Sprintf("%s >= DATEADD(s, %d, '1970-01-01') AND %s <= DATEADD(s, %d, '1970-01-01')", args[0], m.TimeRange.GetFromAsSecondsEpoch(), args[0], m.TimeRange.GetToAsSecondsEpoch()), nil
+
+		return fmt.Sprintf("%s BETWEEN '%s' AND '%s'", args[0], time.Unix(m.TimeRange.GetFromAsSecondsEpoch(), 0), time.Unix(m.TimeRange.GetToAsSecondsEpoch(), 0)), nil
```

Comment:
Please have a look at postgres macros which already implemented the same functionality. You can see that there exists  methods (GetFromAsTimeUTC and GetToAsTimeUTC) on timerange that will give you the time object back, see https://github.com/grafana/grafana/blob/master/pkg/tsdb/postgres/macros.go#L86

I also think it would be good to make the dateformat expected explicit by using .Format()